### PR TITLE
fix(module:select): remove duplicate drop-down menu options when nzMo…

### LINKED ISSUE #3559: nz-select with nzMode='tags' shows duplicate tags
### Reproduction link
[https://angular-btydud.stackblitz.io](https://angular-btydud.stackblitz.io)

### Steps to reproduce
In the select field enter t1 then t2 then t3

### What is expected?
Each item should be shown only once in the dropdown

### What is actually happening?
The dropdown shows each manually entered item two times.

| Environment | Info |
|---|---|
| ng-zorro-antd | 7.5.0 |
| Browser | Chrome Version 74.0.3729.169 (Official Build) (64-bit) |

### PR DESCRIPTION
**Pull request by ZhiDanLuo**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number:#3559

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: components/select/nz-select.service.ts [hunk 1]
```diff
@@ -182,17 +182,20 @@ export class NzSelectService {
       // https://github.com/NG-ZORRO/ng-zorro-antd/issues/3424
       this.listOfTagOption = [...this.listOfCachedSelectedOption, ...this.listOfSelectedValue].reduce(
         (options: NzOptionComponent[], componentOrValue) => {
+          // https://github.com/NG-ZORRO/ng-zorro-antd/issues/3559
           if (
             typeof componentOrValue === 'string' &&
-            !this.listOfTemplateOption.find(o => this.compareWith(o.nzValue, componentOrValue))
+            !this.listOfTemplateOption.find(o => this.compareWith(o.nzValue, componentOrValue)) &&
+            !options.some(o => o.nzValue === componentOrValue)
```
Comment (wenqi73) [inline components/select/nz-select.service.ts]:
```suggestion
            ![...this.listOfTemplateOption, ...options].find(o => o.nzValue === componentOrValue)
```

File: components/select/nz-select.service.spec.ts [hunk 2]
```diff
@@ -86,6 +86,21 @@ describe('SelectService', () => {
...
+      ] as any;
+      service.listOfSelectedValue = [`option_value_a`, `option_value_b`];
+      service.updateListOfTagOption();
+      expect(service.listOfTagOption.length).toEqual(3);
+      expect(service.listOfTagOption[0].nzValue).toEqual('option_value_a');
+      expect(service.listOfTagOption[0].nzLabel).toEqual('option_label_a');
+      expect(service.listOfTagOption[1].nzValue).toEqual('option_value_miss');
+      expect(service.listOfTagOption[1].nzLabel).toEqual('option_label_miss');
+      expect(service.listOfTagOption[2].nzValue).toEqual('option_value_b');
+      expect(service.listOfTagOption[2].nzLabel).toEqual('option_value_b');
```
Comment (wenqi73) [inline components/select/nz-select.service.spec.ts]: 测试项直接加在上面的 `listOfCachedSelectedOption` 数组。

**Comment by wenqi73** [comment]
@ZhiDanLuo Thank you for your PR!

File: components/select/nz-select.service.spec.ts (hunk 2 above)
Comment (ZhiDanLuo) [reply components/select/nz-select.service.spec.ts]: 已修改

**Comment by wenqi73** [comment]
Hi @ZhiDanLuo, we have refactor this solution in #3574. Thx for you PR!

**Pull request closed without merging**